### ISSUE TITLE
Failure to build on macOS 10.13 with Xcode 9

### ISSUE THREAD

**Issue by fxcoudert** [nicklockwood/SwiftFormat#169]
The error, found during Homebrew testing on 10.13, is below. I've seen there is a pull request about the issue, but it was closed. Some I'm filing an issue.

```
CompileSwift normal x86_64
    cd /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4
    /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift -frontend -c /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/Sources/Rules.swift /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/Sources/Tokenizer.swift /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/Sources/Formatter.swift /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/Sources/SwiftFormat.swift /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/Sources/Options.swift /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/Sources/CommandLine.swift /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/CommandLineTool/main.swift -target x86_64-apple-macosx10.11 -enable-objc-interop -sdk /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -I /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/Release -F /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/Release -g -module-cache-path /Users/fx/Library/Developer/Xcode/DerivedData/ModuleCache -swift-version 3 -serialize-debugging-options -Xcc -I/tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/swift-overrides.hmap -Xcc -iquote -Xcc /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/swiftformat-generated-files.hmap -Xcc -I/tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/swiftformat-own-target-headers.hmap -Xcc -I/tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/swiftformat-all-target-headers.hmap -Xcc -iquote -Xcc /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/swiftformat-project-headers.hmap -Xcc -I/tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/Release/include -Xcc -I/tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/DerivedSources/x86_64 -Xcc -I/tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/DerivedSources -Xcc -working-directory/tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4 -emit-module-doc-path /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/Objects-normal/x86_64/swiftformatTool.swiftdoc -serialize-diagnostics-path /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/Objects-normal/x86_64/Rules.dia -O -module-name swiftformatTool -emit-module-path /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/Objects-normal/x86_64/swiftformatTool.swiftmodule -emit-objc-header-path /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/Objects-normal/x86_64/swiftformatTool-Swift.h -emit-dependencies-path /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/Objects-normal/x86_64/Rules.d -num-threads 4 -o /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/Objects-normal/x86_64/Rules.o -o /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/Objects-normal/x86_64/Tokenizer.o -o /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/Objects-normal/x86_64/Formatter.o -o /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/Objects-normal/x86_64/SwiftFormat.o -o /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/Objects-normal/x86_64/Options.o -o /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/Objects-normal/x86_64/CommandLine.o -o /tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/build/SwiftFormat.build/Release/CommandLineTool.build/Objects-normal/x86_64/main.o
/tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/Sources/Rules.swift:2723:52: error: cannot assign value of type 'String.UnicodeScalarView.SubSequence' (aka 'Substring.UnicodeScalarView') to type 'String.UnicodeScalarView'
                characters = string.unicodeScalars.suffix(from: prefix.unicodeScalars.endIndex)
                             ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/Sources/Tokenizer.swift:452:20: error: cannot assign value of type 'String.UnicodeScalarView.SubSequence' (aka 'Substring.UnicodeScalarView') to type 'String.UnicodeScalarView'
            self = suffix(from: index)
                   ^~~~~~~~~~~~~~~~~~~
/tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/Sources/Tokenizer.swift:468:20: error: cannot assign value of type 'String.UnicodeScalarView.SubSequence' (aka 'Substring.UnicodeScalarView') to type 'String.UnicodeScalarView'
            self = suffix(from: index)
                   ^~~~~~~~~~~~~~~~~~~
/tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/Sources/Tokenizer.swift:476:20: error: cannot assign value of type 'String.UnicodeScalarView.SubSequence' (aka 'Substring.UnicodeScalarView') to type 'String.UnicodeScalarView'
            self = dropFirst()
                   ^~~~~~~~~~~
/tmp/swiftformat-20170824-32910-1q2jme4/SwiftFormat-0.29.4/Sources/Tokenizer.swift:484:20: error: cannot assign value of type 'String.UnicodeScalarView.SubSequence' (aka 'Substring.UnicodeScalarView') to type 'String.UnicodeScalarView'
            self = dropFirst()
                   ^~~~~~~~~~~
```

**Comment by nicklockwood** [comment]
@fxcoudert fair enough. I can't land the fix in its current state as it makes SwiftFormat 5000% (50x) slower. Hopefully Apple will address this before Xcode 9 is out of beta.

In the meantime, I suggest using the pre-built binary in this repo instead of homebrew for 10.13.

**Comment by nicklockwood** [comment]
Fixed in 0.29.5
